artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -713,12 +730,16 @@ public void reconnect(AMQConnectionContext existingContext, ConnectionInfo info)
       context.incRefCount();
    }
 
+   private static void traceSendCommand(Command command) {
+      ActiveMQServerLogger.LOGGER.trace("sending " + command);
+   }
+
    /**
     * This will answer with commands to the client
     */
    public boolean sendCommand(final Command command) {
-      if (ActiveMQServerLogger.LOGGER.isTraceEnabled()) {
-         ActiveMQServerLogger.LOGGER.trace("sending " + command);
+      if (ACTIVE_MQ_SERVER_LOGGER_TRACE_ENABLED) {
```

Comment:
I know and it seems an issue with this logger, because it creates a point of contention between all the threads (all in the broker!) because it is locked (I will provide soon a trace of it): I don't think could exist a clean way to fix it other than on the jboss logger and it isn't feasible...

Reply:
Why not fix it at the route in jboss logger? Surely then it be global fix. Im much more happy to fix things at root source.

Reply:
You're quite right, probably I will revert it but it is actually an issue and probably could be turned into a lock-free way: actually it really create a point of contentions for any send of messages, affecting responses and consumers all together...not good!